tests/shellcheck/shellcheck.bzl

```diff
@@ -1,14 +1,16 @@
 load("//tests:inline_tests.bzl", "sh_inline_test")
 
-def shellcheck(name, args, data, sh_flavor="bash", visibility=None):
+def shellcheck(name, args, data, sh_flavor=None, excludes=[], visibility=None):
+  excludes_arg = "" if len(excludes) == 0 else ("--exclude=" + ",".join(excludes))
+  shell_arg = "" if (sh_flavor == None) else ("--shell " + sh_flavor)
   sh_inline_test(
     name = name,
     visibility = visibility,
     size = "small",
     args = args,
     data = data,
     script = """\
-shellcheck --color=always "--shell=%s" $1
-""" % sh_flavor,
+shellcheck --color=always %s %s $1
+""" % (shell_arg, excludes_arg),
```

Comment:
I don't understand this suggestion, where do you want to use a shell array here?

Reply:
Ah, sorry, array_literal is for constructing a bash array.

I meant constructing the command from a list of escaped args instead of concatenating multiple strings:

```
([shell.quote(c) for c in command_list] ++ [ '"$1"' ]).join(" ")
```

where command_list is just a list of command and arguments (and $1 is passed unquoted because the `$` would not be expanded otherwise)

Reply:
Or maybe

```
script = """
{cmd} "$1"
""".format([shell.quote(c) for c in command_list].join(" ")
```

which might be easier to read

Reply:
Isn't the following easier to read:

```
script = """
shellcheck --color=always {} {} "$1"
""".format(shell_arg, excludes_arg),
```

?

Reply:
Well, it’s still only correct by semi-accident and not by construction, but let’s not dwell on it.

Reply:
You mean because the `--color=always` and `{}` arguments are not quoted? If that's you concern, we can quote them or use your suggestion above. Both are fine by me.